qhbmlib/inference/qnn.py

```diff
@@ -91,6 +103,39 @@ def circuit(self):
   def differentiator(self):
     return self._differentiator
 
+  def _sampled_expectation(self, initial_states: tf.Tensor,
+                           observable: hamiltonian.Hamiltonian):
```

Comment:
somehow finding the type `hamiltonian.Hamiltonian` confusing because it's not obvious that this argument needs to be a model. I guess it's related to the import being from `models` instead of importing `models`. But I remember you are following the TFP pattern in doing so. Perhaps, clarifying this in the docstring would help?

Reply:
Not sure what you mean, in the code below `observable.circuit_dagger` and `observable.energy` are called, which only exist if the input is a `Hamiltonian`

Reply:
And Hamiltonian is a specific type of model parameterization - essentially a QHBM. "Hamiltonian" to most folks may mean Hermitian operator specified as matrix or arbitrary parameterization

Reply:
(This concern is not very specific to this PR but the same confusion recurred so wanted to flag for your thoughts)

Reply:
Ok, your original comment was just referring to the name of the type?

Reply:
Yeah what I was saying was that if the type was model.Hamiltonian then this might be clearer. But you didn't import that way because I recall we decided to follow the TFP convention. But honestly not a whole lot clearer compared to a class rename, on second thought.

Reply:
`ModularHamiltonian` sounds great to me!

Reply:
Added new issue #201